src/ui/castlekeysmouse.pas

```diff
@@ -1300,19 +1304,41 @@ function TInputPressRelease.ToString: string;
     Result := Result + ', key repeated';
 end;
 
+function TInputPressRelease.KeyCharacter: char;
+begin
+  {$ifdef MSWINDOWS}
+  { It seems that GTK 1.3 for Windows cannot translate GDK_KEY_Escape and
+    GDK_KEY_Return to standard chars (#13 and #27). So I'm fixing it here. }
+  if Key = K_Escape then
+  Result := CharEscape else
+  if Key = K_Enter then
+  Result := CharEnter else
+  {$endif}
+  { It seems that GTK 2 doesn't translate backspace and tab to
+    appropriate chars. So I'm fixing it here. }
+  if Key = K_Tab then
+  Result := CharTab else
+  if Key = K_BackSpace then
+  Result := CharBackSpace else
+
+  if Length(Self.KeyString) = 1 then
+   Result := Self.KeyString[1] else
+   Result := #0;
```

Comment:
Use just `KeyString`, instead of `Self.KeyString`. Usually qualifying things with `Self.` is not needed, and it does not seem needed here?

Reply:
Yes, I've made it just to "make clear" (after switching paradigm from GTK procedure :)). I'll fix it.